src/App.tsx

```diff
@@ -15,33 +15,109 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export const App: React.FC = () => {
+  const [isOpened, setValue] = useState(false);
+  const [visibleGoods, setValue2] = useState([...goodsFromServer]);
+
+  const startGoods = () => {
+    setValue(true);
+  };
+
+  const reversedGoods = () => {
+    setValue2([...visibleGoods.reverse()]);
+  };
+
+  const sortByABCGoods = () => {
+    setValue2([...visibleGoods.sort((good1, good2) => {
+      return good1.localeCompare(good2);
+    })]);
+  };
+
+  const sortByGoodsLength = () => {
+    setValue2([...visibleGoods.sort((good1, good2) => {
+      return (good1.length - good2.length);
+    })]);
+  };
```

Comment:
do you need to split array after you've already performed actions on it?
you need some fixes here